src/main/java/io/excaliburfrc/robot/subsystems/Drive.java

```diff
@@ -19,15 +43,112 @@
+              DrivetrainConstants.MAX_SPEED_METERS_PER_SECOND,
+              DrivetrainConstants.MAX_ACCELERATION_METERS_PER_SECOND_SQUARED)
+          .setKinematics(driveKinematics)
+          .addConstraint(autoVoltageConstraint);
+  public final Trajectory trajectory =
+      TrajectoryGenerator.generateTrajectory(
+          new Pose2d(0, 0, new Rotation2d(0)),
+          List.of(new Translation2d(1, 1), new Translation2d(2, -1)),
+          new Pose2d(3, 0, new Rotation2d(0)),
+          config);
+
+  private final DifferentialDriveOdometry odometry;
+  private final AHRS ahrs = new AHRS();
+  private final SparkMaxPIDController leftController = leftLeader.getPIDController();
+  private final SparkMaxPIDController rightController = rightLeader.getPIDController();
+  private Rotation2d offset = new Rotation2d(0);
+  private final SimpleMotorFeedforward feedforward = new SimpleMotorFeedforward(kS, kV, kA);
+
   public Drive() {
-    leftFollower.follow(leftLeader);
-    rightFollower.follow(rightLeader);
+    ValidateREVCAN(
+        // reset factory settings
+        leftLeader.restoreFactoryDefaults(),
+        leftFollower.restoreFactoryDefaults(),
+        rightLeader.restoreFactoryDefaults(),
+        rightFollower.restoreFactoryDefaults(),
+        // set the motors to coast mode -- we don't want to break them!
+        leftLeader.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        leftFollower.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        rightLeader.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        rightFollower.setIdleMode(CANSparkMax.IdleMode.kBrake),
+        // have the leader send its applied output as frequently as possible,
+        // to speed up follower response
+        leftLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus0, minimal_FRAME_PERIOD),
+        leftLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus1, minimal_FRAME_PERIOD),
+        leftLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus2, minimal_FRAME_PERIOD),
+        rightLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus0, minimal_FRAME_PERIOD),
+        rightLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus1, minimal_FRAME_PERIOD),
+        rightLeader.setPeriodicFramePeriod(PeriodicFrame.kStatus2, minimal_FRAME_PERIOD),
+        // other status frames can be reduced to almost never
+        leftFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus0, MAXIMAL_FRAME_PERIOD),
+        leftFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus1, MAXIMAL_FRAME_PERIOD),
+        leftFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus2, MAXIMAL_FRAME_PERIOD),
+        rightFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus0, MAXIMAL_FRAME_PERIOD),
+        rightFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus1, MAXIMAL_FRAME_PERIOD),
+        rightFollower.setPeriodicFramePeriod(PeriodicFrame.kStatus2, MAXIMAL_FRAME_PERIOD),
+        // setup following
+        leftFollower.follow(leftLeader),
+        rightFollower.follow(rightLeader));
```

Comment:
maybe this CAN config should be split into a separate PR (along with some other stuff) so that this PR can focus on the odometry/trajectory stuff?